lib/docker.ts

```diff
@@ -102,8 +162,27 @@ export const configureNextcloud = async function () {
 	await runExec(container, ['php', 'occ', 'config:system:set', 'force_locale', '--value', 'en_US'], true)
 	await runExec(container, ['php', 'occ', 'config:system:set', 'enforce_theme', '--value', 'light'], true)
 
-	// Enable the app and give status
-	await runExec(container, ['php', 'occ', 'app:enable', '--force', 'viewer'], true)
+	// Build app list
+	const json = await runExec(container, ['php', 'occ', 'app:list', '--output', 'json'], false)
+	// fix dockerode bug returning invalid leading characters
+	const applist = JSON.parse(json.substring(json.indexOf('{')))
+
+	// Enable apps and give status
+	for (const app of apps) {
+		if (app in applist.enabled) {
+			console.log(`├─ ${app} version ${applist.enabled[app]} already installed and enabled`)
+		} else if (app in applist.disabled) {
+			// built in
+			await runExec(container, ['php', 'occ', 'app:enable', '--force', app], true)
+		} else if (app in VENDOR_APPS) {
+			// some apps are vendored but not within the server package
+			await runExec(container, ['git', 'clone', '--depth=1', VENDOR_APPS[app], `apps/${app}`], true)
```

Comment:
We need to be able to use the apps branch, example given: `stable24`.
if I use this for server on `stable24`, `viewer` would also need to be on `stable24`

Reply:
Good point, fixed :)